include/llvm/Support/WinTypes.h

```diff
@@ -433,28 +433,69 @@ class CSimpleArray : public std::vector<T> {
 #define IsEqualIID(a, b) a==b
 #define IsEqualCLSID(a, b) !memcmp(&a, &b, sizeof(GUID))
 
-class CW2A {
-public:
-  CW2A(const wchar_t *wide, uint32_t utf = 0) {
-    (void)utf;
-    size_t len = wcslen(wide) + 1; // +1 for '\0'
-    m_psz = (char*) malloc(len * sizeof(char));
-    wcstombs(m_psz, wide, len);
-    m_psz[len-1] = '\0';
-  }
-  ~CW2A() {
-    free(m_psz);
+template< int t_nBufferLength = 128 >
+class CW2AEX
+{
+  public:
+  CW2AEX( LPCWSTR psz , UINT nCodePage = CP_UTF8)
+  {
+    if (nCodePage != CP_UTF8) {
+      // you have reached the limits of this half-assed implementation. So sorry.
+      m_psz = NULL;
+      return;
```

Comment:
If it does not cause crashes/errors in the basic use-case (running dxc on an hlsl file), then we should die here, so that we know when some bad usage occurs, rather than silently chasing a nullptr.

```cpp
assert(false && "CW2AEX implementation for Linux does not handle non-UTF8 code pages");
return;
```